src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy

```diff
@@ -686,6 +686,82 @@ class DockerWorkflowFunctionalTest extends AbstractFunctionalTest {
         build('workflow')
     }
 
+    def "Can build an image and save it to file"() {
+        File imageDir = temporaryFolder.newFolder('images', 'minimal')
+        File dockerFile = createDockerfile(imageDir)
+        String imageName = createUniqueImageId()
+        File savedImage = temporaryFolder.newFile()
+
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+            task buildImage(type: DockerBuildImage) {
+                inputDir = file("${dockerFile.parentFile.path}")
+                tag = "${imageName}"
+            }
+
+            task saveImage(type: DockerSaveImage) {
+                dependsOn buildImage
+                repository = "${imageName}"
+                destFile = new File("${savedImage.absolutePath}")
+            }
+
+            task workflow {
+                dependsOn saveImage
+            }
+        """
+
+        expect:
+        build('workflow')
+        savedImage.exists()
+        savedImage.newInputStream().available()
```

Comment:
The `exists()` will always return true since we created the file and the `available()` check _should_ always return true here as there will always be something to read ... I think?

What about removing this test altogether and just focus on the one you wrote below whilst adding additional tasks to stand up a container based on the image we just loaded and check if it's started, has ports open, etc?

Reply:
Yeah, `exists()` is redundant. `anailable()` is used by `docker-java` folks to test their code so I would leave it.

I like the idea of a single test, the workflow might be: build -> save -> remove -> load -> create container.